applications/asset_tracker/src/main.c

```diff
@@ -205,6 +207,36 @@ void error_handler(enum error_type err_type, int err_code)
 #endif /* CONFIG_DEBUG */
 }
 
+void z_SysFatalErrorHandler(unsigned int reason,
+			    const NANO_ESF *pEsf)
+{
+	ARG_UNUSED(pEsf);
+
+#if !defined(CONFIG_SIMPLE_FATAL_ERROR_HANDLER)
+#if defined(CONFIG_STACK_SENTINEL)
+	if (reason == _NANO_ERR_STACK_CHK_FAIL) {
+		goto error_handler;
+	}
+#endif
+	if (reason == _NANO_ERR_KERNEL_PANIC) {
+		goto error_handler;
+	}
+
+	if (z_is_thread_essential()) {
+		printk("Fatal fault in essential thread! Spinning...\n");
```

Comment:
That message might be confusing when we reboot. I'd either remove it, or move to the `error_handler` where we actually could spin.

Reply:
I've decided to remove the "Spinning" message but keep the information that an essential thread has failed.